### DISCUSSION TITLE
What about performance of Polygon Mesh Processing Library  relative to OpenMesh Library

### DISCUSSION THREAD

**Discussion by geotyper** [pmp-library/pmp-library#154] [General]
What about performance of Polygon Mesh Processing Library  relative to OpenMesh Library?
use OpenMesh for some hobby projects and main problem that OpenMesh work slow when need to add-delete some faces.
Now think about migrate to PMP or not,  also see that OpenMesh is live project and have regular updates. Is any performance increase in PMP, or it only more friendly designed to user?
and one more question: is PMP have some flags for activate all safety checks, like NGC_SAFETY_CHECKS in Geometry Central?

**Comment by dsieger** [comment]
Good question! When we did the original version of the SurfaceMesh data structure we did extensive comparisons with the existing data structures at the time. You can find the original paper [here](https://ls7-gv.cs.tu-dortmund.de/publications/2011/imr11/imr11.html). Of course, that was more than 10 years ago! I did not do any recent benchmarking, so I don't know if there are significant performance differences in the current implementations

PMP has some additional assertions when compiling in debug mode, but that's all. Unfortunately, I don't know much about GeometryCentral, so I can't compare.

Hope this helps!
